content/kb/tutorials/deploy/docker.md

```diff
@@ -0,0 +1,236 @@
+
+- **On your corporate network** - Most corporate networks are closed to the outside world. You typically use a VPN to log onto your corporate network and access resources there. You could run your Streamlit app on a server in your corporate network for security reasons, to ensure that only folks internal to your company can access it.
+- **On the cloud** - If you'd like to access your Streamlit app from outside of a corporate network, or share your app with folks outside of your home network or laptop, you might choose this option. In this case, it'll depend on your hosting provider. We have community-submitted guides from Heroku, AWS, and other providers [here](/knowledge-base/deploy/deploy-streamlit-heroku-aws-google-cloud).
+
+Wherever you decide to deploy your app, you will first need to containerize it. This guide walks you through using Docker to deploy your app. If you prefer Kubernetes see [Deploy Streamlit using Kubernetes](/knowledge-base/tutorials/deploy/kubernetes).
+
+## Prerequisites
+
+1. [Install Docker Engine](#install-docker-engine)
+2. [Check network port accessibility](#check-network-port-accessibility)
+
+### Install Docker Engine
+
+If you haven't already done so, install [Docker](https://docs.docker.com/engine/install/#server) on your server. Docker provides `.deb` and `.rpm` packages from many Linux distributions, including:
+
+- [Debian](https://docs.docker.com/engine/install/debian/)
+- [Ubuntu](https://docs.docker.com/engine/install/ubuntu/)
+
+Verify that Docker Engine is installed correctly by running the `hello-world` Docker image:
+
+```bash
+sudo docker run hello-world
+```
+
+<Tip>
+
+Follow Docker's official [post-installation steps for Linux](https://docs.docker.com/engine/install/linux-postinstall/) to run Docker as a non-root user, so that you don't have to preface the `docker` command with `sudo`.
+
+</Tip>
+
+### Check network port accessibility
+
+As you and your users are behind your corporate VPN, you need to make sure all of you can access a certain network port. Let's say port `8501`, as it is the default port used by Streamlit. Contact your IT team and request access to port `8501` for you and your users.
+
+## Create a Dockerfile
+
+Docker builds images by reading the instructions from a `Dockerfile`. A `Dockerfile` is a text document that contains all the commands a user could call on the command line to assemble an image. Click [here](https://docs.docker.com/engine/reference/builder/) to learn more. The [docker build](https://docs.docker.com/engine/reference/commandline/build/) command builds an image from a `Dockerfile`. The [docker run](https://docs.docker.com/engine/reference/commandline/run/) command first creates a container over the specified image, and then starts it using the specified command.
```

Comment:
Can you use meaningful [link text](https://developers.google.com/style/link-text)?